stl/inc/span

```diff
@@ -0,0 +1,679 @@
+        }
+
+        if (_Off < 0) {
+            _STL_VERIFY(_Myend - _Myptr >= -_Off, "cannot seek span iterator after end");
+        }
+#endif // _ITERATOR_DEBUG_LEVEL >= 1
+        _Myptr -= _Off;
+        return *this;
+    }
+
+    _NODISCARD constexpr _Span_iterator operator-(const difference_type _Off) const noexcept {
+        _Span_iterator _Tmp{*this};
+        _Tmp -= _Off;
+        return _Tmp;
+    }
+
+    _NODISCARD constexpr difference_type operator-(const _Span_iterator& _Right) const noexcept {
+#if _ITERATOR_DEBUG_LEVEL >= 1
+        _STL_VERIFY(
+            _Mybegin == _Right._Mybegin && _Myend == _Right._Myend, "cannot subtract incompatible span iterators");
+#endif // _ITERATOR_DEBUG_LEVEL >= 1
+        return _Myptr - _Right._Myptr;
+    }
+
+    _NODISCARD constexpr reference operator[](const difference_type _Off) const noexcept {
+        return *(*this + _Off);
+    }
+
+    _NODISCARD constexpr bool operator==(const _Span_iterator& _Right) const noexcept {
+#if _ITERATOR_DEBUG_LEVEL >= 1
+        _STL_VERIFY(_Mybegin == _Right._Mybegin && _Myend == _Right._Myend,
+            "cannot compare incompatible span iterators for equality");
+#endif // _ITERATOR_DEBUG_LEVEL >= 1
+        return _Myptr == _Right._Myptr;
+    }
+
+    _NODISCARD constexpr bool operator!=(const _Span_iterator& _Right) const noexcept {
+        return !(*this == _Right);
+    }
+
+    _NODISCARD constexpr bool operator<(const _Span_iterator& _Right) const noexcept {
+#if _ITERATOR_DEBUG_LEVEL >= 1
+        _STL_VERIFY(
+            _Mybegin == _Right._Mybegin && _Myend == _Right._Myend, "cannot compare incompatible span iterators");
+#endif // _ITERATOR_DEBUG_LEVEL >= 1
+        return _Myptr < _Right._Myptr;
+    }
+
+    _NODISCARD constexpr bool operator>(const _Span_iterator& _Right) const noexcept {
+        return _Right < *this;
+    }
+
+    _NODISCARD constexpr bool operator<=(const _Span_iterator& _Right) const noexcept {
+        return !(_Right < *this);
+    }
+
+    _NODISCARD constexpr bool operator>=(const _Span_iterator& _Right) const noexcept {
+        return !(*this < _Right);
+    }
+
+#if _ITERATOR_DEBUG_LEVEL >= 1
+    friend constexpr void _Verify_range(const _Span_iterator& _First, const _Span_iterator& _Last) {
+        _STL_VERIFY(_First._Mybegin == _Last._Mybegin && _First._Myend == _Last._Myend,
+            "span iterators from different views do not form a range");
+        _STL_VERIFY(_First._Myptr <= _Last._Myptr, "span iterator range transposed");
+    }
+#endif // _ITERATOR_DEBUG_LEVEL >= 1
+
+    using _Prevent_inheriting_unwrap = _Span_iterator;
+
+    _NODISCARD constexpr pointer _Unwrapped() const noexcept {
+        return _Myptr;
+    }
+
+    static constexpr bool _Unwrap_when_unverified = _ITERATOR_DEBUG_LEVEL == 0;
+
+    constexpr void _Seek_to(const pointer _It) noexcept {
+        _Myptr = _It;
+    }
+
+#if _ITERATOR_DEBUG_LEVEL >= 1

```

Comment:
I have a mild preference to put these 2 members after `_Myptr` so that a reinterpret_cast from IDL1/2 to IDL0 world still gives a reasonable answer. Also that lets optimized _IDL1/2 be a no-op/identity conversion since the address of this and _Myptr would match.

Reply:
Putting `_Myptr` first makes sense for codegen (offset 0). I am not persuaded by IDL cross-compatibility, that is always doomed.

Reply:
Done